File: server/src/main/java/financial/management/com/server/models/FinancialRelease.java
package financial.management.com.server.models;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.json.JSONObject;
import org.springframework.data.annotation.Id;
import financial.management.com.server.enums.FinancialReleaseType;
import org.springframework.data.mongodb.core.mapping.Document;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Document(collection = "financialreleases")
public class FinancialRelease {

    @Id
    private String id;
    private String name;
    private Float value;
    private FinancialReleaseType type;
    private Integer year;
    private Integer month;
    private Integer day;
}

File: client/src/app/models/FinancialRelease.ts
import { FinancialReleaseType } from './FinancialReleaseType';
import { Person } from './Person';

export class FinancialRelease {
  id: string | null
  name: string;
  value: number;
  type: FinancialReleaseType;
  person: Person;
  year: number;
  month: number;
  day: number;

  date?: Date;

  constructor(
    id: string | null,
    name: string,
    value: number,
    type: FinancialReleaseType,
    person: Person,
    year: number,
    month: number,
    day: number | null
  ) {
    this.id = id;
    this.name = name;
    this.value = value;
    this.type = type;
    this.person = person;
    this.year = year;
    this.month = month;
    this.day = day || 1;
  }
}
